### PR TITLE
fix: `5.3 Manual Recovery With Receive Call Does Not Work`

### PR DESCRIPTION
**Pull request by zerotucks**
## Summary by CodeRabbit

- New Features
  - Router-level failure recording, queryable failure records, and governor-controlled retry; router exposes a sweep recovery API.

- Changes
  - Failures during delivery are recorded (no longer revert) and surfaced via new events/getters.
  - Public API renamed from "recover assets" to "sweep".

- Refactor
  - Adapter-level manual recovery and local failure tracking removed; recovery centralized in the router.

- Tests
  - Tests added/updated for recorded-failure semantics, retry flows, sweep, and edge cases (native/ERC20, reentrancy, overrides).

### CODE REVIEW

File: packages/chain-bridge/src/router/BridgeRouter.sol [hunk 1]
```diff
@@ -606,6 +775,282 @@ contract BridgeRouter is
...
+                originalPayload,
+                (BridgeTypes.RelayedTransferParams)
+            );
+
+            // Apply recipient override only
+            // Asset addresses are handled by adapters for cross-chain compatibility
+            if (overrides.recipient != address(0)) {
+                params.recipient = overrides.recipient;
+            }
+
```
Comment (halaprix) [inline packages/chain-bridge/src/router/BridgeRouter.sol:888]: this check is redundant - since the `receipient` is the only paramater that can be overriden - hence if overrrideData length is > 0 - then it's the receipient

**Review by halaprix — CHANGES_REQUESTED** [review]

File: packages/chain-bridge/src/router/BridgeRouter.sol [hunk 2]
```diff
@@ -685,6 +847,218 @@ contract BridgeRouter is
...
+        BridgeTypes.OperationType operationType,
+        bytes memory originalPayload,
+        address newRecipient
+    ) internal pure returns (bytes memory modifiedPayload) {
+        if (operationType == BridgeTypes.OperationType.TRANSFER_ASSET) {
+            BridgeTypes.RelayedTransferParams memory params = abi.decode(
+                originalPayload,
+                (BridgeTypes.RelayedTransferParams)
+            );
+
```
Comment (halaprix) [inline packages/chain-bridge/src/router/BridgeRouter.sol:940]: you can use `_validateArkFleetRelationship` here and get rid of bulk amount of code -> `_validateRetryPayload` , `_validateTransferPayload`, `_validateMessagePayload`

**Review by halaprix — CHANGES_REQUESTED** [review]
I've left one comment and the gtg

**Review by halaprix — APPROVED** [review]
